### ISSUE TITLE
lona-server crashes on user-input

### ISSUE THREAD

**Issue by SmithChart** [lona-web-org/lona#217]
On current master I can crash the lona-server by entering anything into a text-input in the browser.

My minimal reproducer is:
```
#!/usr/bin/env python3

from datetime import datetime
from lona.html import HTML, TextInput, Button
from lona import LonaApp, LonaView

app = LonaApp(__file__)

@app.route('/')
class ClockView(LonaView):
    def handle_request(self, request):
        text = TextInput()
        button = Button("Click here")
        html = HTML(
            text,
            button,
        )

        while True:
            self.await_input_event(html=html)

app.run()
```

I am running this with lona on dd49788e64cb2cd02e7a33e1f973eb4409bb7302 (master).
But it's working as expected on 1.9.

Steps to reproduce:
* Start the reproducer
* Access localhost:8080
* Click the button. Nothing happens.
* Enter anything into the text box.
* Click the button again.
* I get the following stack trace in the server console:

```
======== Running on http://localhost:8080 ========
(Press CTRL+C to quit)
MainThread                     INFO     21:59:00.879650 aiohttp.access ::1 [08/Mar/2022:20:59:00 +0000] "GET / HTTP/1.1" 200 320 "-" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/99.0.4844.51 Safari/537.36"
MainThread                     INFO     21:59:00.942757 aiohttp.access ::1 [08/Mar/2022:20:59:00 +0000] "GET / HTTP/1.1" 200 874 "http://localhost:8080/" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/99.0.4844.51 Safari/537.36"
MainThread                     INFO     21:59:00.943334 aiohttp.access ::1 [08/Mar/2022:20:59:00 +0000] "GET /favicon.ico HTTP/1.1" 200 231 "http://localhost:8080/" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/99.0.4844.51 Safari/537.36"
MainThread                     INFO     21:59:00.963651 aiohttp.access ::1 [08/Mar/2022:20:59:00 +0000] "GET /static/lona/style.css HTTP/1.1" 200 231 "http://localhost:8080/" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/99.0.4844.51 Safari/537.36"
MainThread                     INFO     21:59:00.964097 aiohttp.access ::1 [08/Mar/2022:20:59:00 +0000] "GET /static/_lona/lona.js HTTP/1.1" 200 250 "http://localhost:8080/" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/99.0.4844.51 Safari/537.36"
MainThread                     INFO     21:59:00.985878 aiohttp.access ::1 [08/Mar/2022:20:59:00 +0000] "GET /favicon.ico HTTP/1.1" 200 231 "http://localhost:8080/" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/99.0.4844.51 Safari/537.36"
LonaRuntimeWorker_1            ERROR    21:59:08.958428 lona.view_runtime Exception raised while running <__main__.ClockView object at 0x7fdea424d8e0>
  Traceback (most recent call last):
    File "/home/chris/work/Projects/github/lona/lona/view_runtime.py", line 318, in start
      raw_response_dict = self.view.handle_request(self.request) or ''
    File "/home/chris/Projekte/ongoing/lona-infobeamer/./reproducer.py", line 22, in handle_request
      self.await_input_event(html=html)
    File "/home/chris/work/Projects/github/lona/lona/view.py", line 228, in await_input_event
      return self._await_specific_input_event(
    File "/home/chris/work/Projects/github/lona/lona/view.py", line 209, in _await_specific_input_event
      self.show(html=html)
    File "/home/chris/work/Projects/github/lona/lona/view.py", line 143, in show
      self._view_runtime.send_data(
    File "/home/chris/work/Projects/github/lona/lona/view_runtime.py", line 546, in send_data
      return self._send_html_update(
    File "/home/chris/work/Projects/github/lona/lona/view_runtime.py", line 535, in _send_html_update
      message = encode_data(
    File "/home/chris/work/Projects/github/lona/lona/protocol.py", line 213, in encode_data
      return PROTOCOL.MESSAGE_PREFIX.value + dumps(
    File "/home/chris/work/Projects/github/lona/lona/_json.py", line 20, in dumps
      return json.dumps(*args, default=default, separators=separators, **kwargs)
    File "/usr/lib/python3.9/json/__init__.py", line 234, in dumps
      return cls(
    File "/usr/lib/python3.9/json/encoder.py", line 199, in encode
      chunks = self.iterencode(o, _one_shot=True)
    File "/usr/lib/python3.9/json/encoder.py", line 257, in iterencode
      return _iterencode(o, 0)
    File "/home/chris/work/Projects/github/lona/lona/_json.py", line 16, in default
      raise TypeError
  TypeError
MainThread                     ERROR    21:59:08.962276 asyncio Future exception was never retrieved
future: <Future finished exception=TypeError()>
  Traceback (most recent call last):
    File "/home/chris/work/Projects/github/lona/lona/view_runtime.py", line 318, in start
      raw_response_dict = self.view.handle_request(self.request) or ''
    File "/home/chris/Projekte/ongoing/lona-infobeamer/./reproducer.py", line 22, in handle_request
      self.await_input_event(html=html)
    File "/home/chris/work/Projects/github/lona/lona/view.py", line 228, in await_input_event
      return self._await_specific_input_event(
    File "/home/chris/work/Projects/github/lona/lona/view.py", line 209, in _await_specific_input_event
      self.show(html=html)
    File "/home/chris/work/Projects/github/lona/lona/view.py", line 143, in show
      self._view_runtime.send_data(
    File "/home/chris/work/Projects/github/lona/lona/view_runtime.py", line 546, in send_data
      return self._send_html_update(
    File "/home/chris/work/Projects/github/lona/lona/view_runtime.py", line 535, in _send_html_update
      message = encode_data(
    File "/home/chris/work/Projects/github/lona/lona/protocol.py", line 213, in encode_data
      return PROTOCOL.MESSAGE_PREFIX.value + dumps(
    File "/home/chris/work/Projects/github/lona/lona/_json.py", line 20, in dumps
      return json.dumps(*args, default=default, separators=separators, **kwargs)
    File "/usr/lib/python3.9/json/__init__.py", line 234, in dumps
      return cls(
    File "/usr/lib/python3.9/json/encoder.py", line 199, in encode
      chunks = self.iterencode(o, _one_shot=True)
    File "/usr/lib/python3.9/json/encoder.py", line 257, in iterencode
      return _iterencode(o, 0)
    File "/home/chris/work/Projects/github/lona/lona/_json.py", line 16, in default
      raise TypeError
  TypeError

  During handling of the above exception, another exception occurred:

  Traceback (most recent call last):
    File "/usr/lib/python3.9/concurrent/futures/thread.py", line 52, in run
      result = self.fn(*self.args, **self.kwargs)
    File "/home/chris/work/Projects/github/lona/lona/server.py", line 549, in _function
      return function(*args, **kwargs)
    File "/home/chris/work/Projects/github/lona/lona/view_runtime.py", line 377, in start
      return self.run_error_view(
    File "/home/chris/work/Projects/github/lona/lona/view_runtime.py", line 177, in run_error_view
      response_dict = self.handle_raw_response_dict(
    File "/home/chris/work/Projects/github/lona/lona/view_runtime.py", line 625, in handle_raw_response_dict
      self.send_data(
    File "/home/chris/work/Projects/github/lona/lona/view_runtime.py", line 560, in send_data
      message = encode_data(
    File "/home/chris/work/Projects/github/lona/lona/protocol.py", line 213, in encode_data
      return PROTOCOL.MESSAGE_PREFIX.value + dumps(
    File "/home/chris/work/Projects/github/lona/lona/_json.py", line 20, in dumps
      return json.dumps(*args, default=default, separators=separators, **kwargs)
    File "/usr/lib/python3.9/json/__init__.py", line 234, in dumps
      return cls(
    File "/usr/lib/python3.9/json/encoder.py", line 199, in encode
      chunks = self.iterencode(o, _one_shot=True)
    File "/usr/lib/python3.9/json/encoder.py", line 257, in iterencode
      return _iterencode(o, 0)
    File "/home/chris/work/Projects/github/lona/lona/_json.py", line 16, in default
      raise TypeError
  TypeError
^CMainThread                     INFO     21:59:09.973428 aiohttp.access ::1 [08/Mar/2022:20:59:00 +0000] "GET / HTTP/1.1" 101 0 "-" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/99.0.4844.51 Safari/537.36"
```

I am not really sure what is happening here so I hope reporting it helps fixing it in master before the next release.

**Comment by fscherf** [comment]
Hi @SmithChart,

Thanks for reporting, i noticed this too. I suspect i broke the protocol layer with this patch ff7a0ab8316c9a4faf96e939d656fc52e66a514d. I will fix this before 1.10 gets released.

**Comment by fscherf** [comment]
@SmithChart I found the bug. It was the patch i suspected. Could you give some feedback if this fixes the issue for you?
